tests/robustness/crash_consistency_test/crash_consistency_test.go

```diff
@@ -0,0 +1,262 @@
+//go:build darwin || (linux && amd64)
+// +build darwin linux,amd64
+
+package consistency
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/kopia/kopia/tests/recovery/blobmanipulator"
+	"github.com/kopia/kopia/tests/tools/kopiarunner"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConsistencyWhenKill9AfterModify(t *testing.T) {
+	// create, connect repository
+	dataRepoPath := path.Join(*repoPathPrefix, dirPath+dataPath)
+	baseDir := makeBaseDir()
+	bm, err := blobmanipulator.NewBlobManipulator(baseDir, dataRepoPath)
+	if err != nil {
+		if errors.Is(err, kopiarunner.ErrExeVariableNotSet) {
+			log.Println("Skipping recovery tests because KOPIA_EXE is not set")
+		} else {
+			log.Println("Error creating Blob Manipulator:", err)
+		}
+		os.Exit(0)
+	}
+
+	bm.DataRepoPath = dataRepoPath
+
+	// generate 10M * 10 dataset
+	fileSize := 1 * 1024 * 1024
+	numFiles := 100
+	err = bm.GenerateRandomFiles(fileSize, numFiles)
+	if err != nil {
+		log.Println("Error in creating snapshot data", err)
+		t.FailNow()
+	}
+
+	// create snapshot successfully at the first time
+	// populate the kopia repo under test with random snapshots
+	snapID, _, err := bm.TakeSnapshot(bm.PathToTakeSnapshot)
+	if err != nil {
+		log.Println("Error in creating snapshot", err)
+		t.FailNow()
+	}
+	dst := getRootDir(bm.PathToTakeSnapshot)
+
+	preResotrePath := filepath.Join(dirPath, "restore_pre")
+	// try to restore a snapshot named restore_pre
+	stdout, err := bm.RestoreGivenOrRandomSnapshot("", preResotrePath)
+	if err != nil {
+		log.Println("Error restoring the kopia repository:", stdout, err)
+		t.FailNow()
+	}
```

Comment:
Hmmm. I am trying to understand this step a little better. What do you think is the purpose of trying to perform a restore before the second snapshot ?

I think you maybe have some valid reason here, I just would like to understand it better.